src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/LockFileUtils.cs

```diff
@@ -956,11 +981,50 @@ public static void ExcludeItems(LockFileTargetLibrary lockFileLib, LibraryInclud
                 lockFileLib.ContentFiles.Add(ContentFileUtils.CreateEmptyItem());
             }
 
-            if ((dependencyType & LibraryIncludeFlags.Build) == LibraryIncludeFlags.None)
+            if ((dependencyType & LibraryIncludeFlags.BuildTransitive) == LibraryIncludeFlags.None &&
+                (dependencyType & LibraryIncludeFlags.Build) == LibraryIncludeFlags.None)
```

Comment:
You seem to have a test here. 

https://github.com/NuGet/NuGet.Client/pull/2667/files#diff-b5fc9c4873b83ec596c278b4b6a7fb93R268 and I found that you were handling that case here https://github.com/NuGet/NuGet.Client/pull/2667/files#diff-d43b8c1e0e4c5dbc97ca65932c8856f7R60. 

Disregard the above concern, but I'd leave comment so that it's easier to follow.